pkg/apb/provision.go

```diff
@@ -45,26 +45,36 @@ func Provision(
 		return "", nil, fmt.Errorf("Project %s does not exist", ns)
 	}
 
-	podName, err := ExecuteApb(
+	executionContext, err := ExecuteApb(
 		"provision", clusterConfig, instance.Spec,
 		instance.Context, instance.Parameters, log,
 	)
 
 	if err != nil {
-		log.Errorf("Problem executing apb [%s]", podName)
+		log.Errorf("Problem executing apb [%s]", executionContext.PodName)
 		log.Error(err.Error())
-		return podName, nil, err
+		return executionContext.PodName, nil, err
 	}
 
-	creds, err := ExtractCredentials(podName, instance.Context.Namespace, log)
+	creds, err := ExtractCredentials(executionContext.PodName, executionContext.Namespace, log)
```

Comment:
It's weird to have an instance.Context and this executionContext mixed in the same method.